ruby_oop/mastermind/game.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+require './computer_player'
+require './human_player'
+
+# class to play the mastermind game
+class Game
+  attr_accessor :human_player, :computer_player
+
+  AVAILABLE_COLORS = ['🟤', '🔴', '🔵', '🟢', '🟡', '🟣'].freeze
+  @secret_code = []
+
+  def initialize
+    @human_player = codebreaker_or_codemaker
+    @computer_player = !@human_player
+  end
+
+  def computer_codemaker
+    puts "Computer is selecting random colors...\n\n"
+    @secret_code = AVAILABLE_COLORS.sample(4)
+    ComputerPlayer.new(@secret_code)
+  end
+
+  def computer_codebreaker
+    puts "Computer is selecting random colors...\n\n"
+    @guess_array = AVAILABLE_COLORS.sample(4)
+    ComputerPlayer.new(@guess_array)
+  end
+
+  def human_codemaker
+    puts 'Enter a code to be broken: '
+    @secret_code = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+      pick_color(color_choice_selection), pick_color(color_choice_selection)]
+      HumanPlayer.new(@secret_code)
+  end
+
+  def human_codebreaker
+    @guess_array = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+                    pick_color(color_choice_selection), pick_color(color_choice_selection)]
+    if @guess_array.include?(nil) || @guess_array.uniq.length != 4
+      puts 'Duplicate entries or empty entries not allowed. Enter a new selection again.'
+      @guess_array = [pick_color(color_choice_selection), pick_color(color_choice_selection),
+                      pick_color(color_choice_selection), pick_color(color_choice_selection)]
+    end
+    puts "\nYou guessed: #{@guess_array}"
+    HumanPlayer.new(@guess_array)
+  end
+
+  def codebreaker_or_codemaker
```

Comment:
I find the way this method name odd. It returns either `true` or `false`. `true` means codemaker, but I had to read the implementation to understand that. Also it's Ruby convention to name methods that return a boolean with a `?` prefix. Those methods should ultimately just return `true` or `false`, not also do things like `puts` statements. Otherwise it's surprising for Rubyists.

Ultimately, I think this one method is doing too much, so I recommend splitting up its responsibilities somehow.

Reply:
@chmurph2 based off the current method, is it better or should I continue to refactor? I did move some of the puts outside of the method...